File: backend/FinTech/Dominio/FinTech.Domain/Entidades/Transaction.cs
using FinTech.Domain.Enumeraciones;

namespace FinTech.Domain.Entidades;

public class Transaction
{
    public Guid Id { get; set; }
    public string IdempotencyKey { get; set; } = string.Empty;
    public TransactionType Type { get; set; }
    public decimal Amount { get; set; }
    public TransactionStatus Status { get; set; } = TransactionStatus.Pending;
    public Guid? LoanId { get; set; }
    public string Description { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Loan? Loan { get; set; }
}

File: frontend/features/transactions/types/transaction.ts
export interface TransactionResponse {
  id: string;
  idempotencyKey: string;
  type: string;
  amount: number;
  status: string;
  loanId: string | null;
  description: string;
  createdAt: string;
}

  export interface CreateTransactionRequest {
    idempotencyKey: string;
    type: string;
    amount: number;
    loanId?: string;
    description: string;
  }
